app/gameobjects.ts

```diff
@@ -0,0 +1,67 @@
+import * as wire from './wire';
+import RobotProxy from './robotproxy';
+import {GameState} from './gamestate';
+
+export class GameObject {
+    constructor(public location: wire.Point) {}
+    inRange(other: {location: wire.Point}, range: number) {
```

Comment:
Because the check for `getDistance() <= range` would be done at literally every call site, so why have them separate? Note how there's now need for a `getDistance()` function at present, since all we ever want to do with a distance is check if something is within range.